spinta/manifests/open_api/helpers.py

```diff
@@ -2,99 +2,325 @@
-        'description': info.get('description', ''),
-    }
+    yield (
+        None,
+        {
+            "type": "ns",
+            "name": dataset_prefix,
+            "title": info.get("summary", title),
+            "description": info.get("description", ""),
+        },
+    )
 
 
 def get_resource_parameters(parameters: list[dict]) -> dict[str, dict]:
     resource_parameters = {}
     for index, value in enumerate(parameters):
-        name = value['name']
-        location = value['in'] if value['in'] in SUPPORTED_PARAMETER_LOCATIONS else ''
-        resource_parameters[f'parameter_{index}'] = {
-            'name': to_code_name(name),
-            'source': [name],
-            'prepare': [Expr(location)],
-            'type': 'param',
-            'description': value.get('description', ''),
+        name = value["name"]
+        location = value["in"] if value["in"] in SUPPORTED_PARAMETER_LOCATIONS else ""
+        resource_parameters[f"parameter_{index}"] = {
+            "name": to_code_name(name),
+            "source": [name],
+            "prepare": [Expr(location)],
+            "type": "param",
+            "description": value.get("description", ""),
         }
 
     return resource_parameters
 
 
-def get_dataset_schemas(data: dict, dataset_prefix: str) -> Generator[tuple[None, dict]]:
+model_deduplicator = Deduplicator()
+
+class Model:
+    def __init__(
+        self,
+        dataset: str,
+        resource: str,
+        basename: str,
+        json_schema: dict,
+        source: str | None = None,
+        parent: Model | None = None
+    ) -> None:
+        self.dataset: str = dataset
+        self.resource: str = resource
+        self.basename: str = basename
+        self.json_schema: dict = json_schema
+        self.source: str = source if source is not None else basename
+        self.title: str = self.json_schema.get("title")
+        self.description: str = self.json_schema.get("description")
+        self.name: str = model_deduplicator(f"{self.dataset}/{to_model_name(self.basename)}")
+        self.children: list[Model] = []
+        self.properties: list[Property] = []
+        self.parent = parent
+        self.add_properties()
+
+    def add_properties(self) -> None:
+        for property_name, json_schema in self.json_schema.get("properties", {}).items():
+            prop = self.add_property(property_name, json_schema)
+
+            if json_schema.get("type") == "object":
+                self._add_ref_model(prop)
+
+            elif json_schema.get("type") == "array":
+                json_schema = json_schema.get("items", {})
+                if json_schema.get("type") == "object":
+                    self._add_backref_model(prop)
+                else:
+                    prop = self.add_property(property_name, json_schema, source="")
+                    prop.name += "[]"
+
+    def add_property(self, name:str, json_schema:dict, **kwargs) -> Property:
```

Comment:
Spaces after colons in typing (for the whole file, not just this one place)
```suggestion
    def add_property(self, name: str, json_schema: dict, **kwargs: Any) -> Property:
```